src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.NameGenerator.cs

```diff
@@ -57,6 +57,17 @@ private static ImmutableArray<Words> GetInterleavedPatterns(ArrayBuilder<TextSpa
                     result.Add(GetLongestBackwardSubsequence(length, breaks, baseName, pluralize));
                 }
 
+                for (var index = result.Count - 1; index >= 0; index--)
+                {
+                    // Every words-struct in result corresponds to one name that will be generated.
+                    // The first string inside the words-struct is the start of the name.
+                    // Names shouldn't start with a digit, so results with a digit as the first character of the first word are removed.
+                    if (char.IsDigit(result[index][0], index: 0))
```

Comment:
There are other types of identifier part characters that are not legal identifier start characters. Should we be using `IsValidIdentifier` instead?

https://github.com/dotnet/roslyn/blob/1b4b5ba7e615e8fdbde036806bac710977e0239c/src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs#L443-L447

If doing the parse on each is too slow or something, then we could use `IsIdentifierStartCharacter` on the first character instead of checking exclusively for digits.

Reply:
You are right. I changed it to using `IsValidIdentifier`. This way there won't be a name suggestions that is invalid. I was unable to detect any significant slow down of the declaration name completion feature, using these changes.